File: src/main/java/com/sdp/project/models/volunteer/SkillsDecorator.java
package com.sdp.project.models.volunteer;

public abstract class SkillsDecorator implements IVolunteer {
    protected IVolunteer volunteer;

    public SkillsDecorator(IVolunteer volunteer) {
        this.volunteer = volunteer;
    }

    @Override
    public void addSkill(String skill) {
        volunteer.addSkill(skill);
    }
}

File: models/Volunteer/SkillsDecorator.php
<?php


abstract class SkillsDecorator implements IVolunteer {
    protected IVolunteer $volunteer;

    public function __construct(IVolunteer $volunteer) {
        $this->volunteer = $volunteer;
    }

    public function addSkill(string $skill): void {
        $this->volunteer->addSkill($skill);
    }
}


?>